### ISSUE TITLE
Why the `PointConv` code can't match Fig.4? And what does the following code mean?

### ISSUE THREAD

**Issue by Wen-ZhiCheng** [dylanwu711/PointConvFormer_Clean#1]
what it means?
`        n_alpha = gathered_norm   # [B, S, k, 3]
        n_miu = sparse_xyz_norm   # [B, S, 3]
        r_miu = localized_xyz     # [B, S, k, 3]
        r_hat = F.normalize(r_miu, dim=3)   # [B, S, k, 3]
        # [B, S, k, 3]
        v_miu = n_miu.unsqueeze(dim=2) - torch.matmul(n_miu.unsqueeze(dim=2), r_hat.permute(0, 1, 3, 2)).permute(0, 1, 3, 2) * r_hat
        v_miu = F.normalize(v_miu, dim=3)   # [B, S, k, 3]
        w_miu = torch.cross(r_hat, v_miu, dim=3)   # [B, S, k, 3]
        w_miu = F.normalize(w_miu, dim=3)   # [B, S, k, 3]
        theta1 = torch.matmul(n_alpha, n_miu.unsqueeze(dim=3))    # [B, S, k, 1]
        theta2 = torch.matmul(r_hat, n_miu.unsqueeze(dim=3))      # [B, S, k, 1]
        theta3 = torch.sum(r_hat * n_alpha, dim=3, keepdim=True)  # [B, S, k, 1]
        theta4 = torch.matmul(r_miu, n_miu.unsqueeze(dim=3))      # [B, S, k, 1]
        theta5 = torch.sum(n_alpha * r_hat, dim=3, keepdim=True)  # [B, S, k, 1]
        theta6 = torch.sum(n_alpha * v_miu, dim=3, keepdim=True)  # [B, S, k, 1]
        theta7 = torch.sum(n_alpha * w_miu, dim=3, keepdim=True)  # [B, S, k, 1]
        # [B, S, k, 1]
        theta8 = torch.sum(r_miu * torch.cross(n_alpha, n_miu.unsqueeze(dim=2).repeat(1, 1, K, 1), dim=3), dim=3, keepdim=True)
        theta9 = torch.norm(r_miu, dim=3, keepdim=True)  # [B, S, k, 1]
        # [B, S, k, 12]
        weightNetInput = torch.cat([theta1, theta2, theta3, theta4, theta5, theta6, theta7, theta8, theta9, localized_xyz], dim=3).contiguous()
`
